backend/src/index.js

```diff
@@ -8,11 +8,17 @@
 const PORT = process.env.PORT || 3000;
 
 const app = require( 'express' )();
+const { promisify } = require( 'util' );
 const { Server } = require( 'http' );
 const http = new Server( app );
 const io = require( 'socket.io' )( http );
+const Redis = require( 'ioredis' );
+const redisAdapter = require( 'socket.io-redis' );
+io.adapter( redisAdapter( { host: 'localhost', port: 6379 } ) );
 
-const db = new Map();
+const client = new Redis();
+const fetchUsers = promisify( client.get ).bind( client );
+const storeUsers = promisify( client.set ).bind( client );
```

Comment:
IMO these names are not the best. They suggest you operate on objects when you operate on plain strings. I think the best option is to create separate functions and make serialization/deserialization (stringify/parse) in them.

Reply:
I've removed it. These functions support promises by default. I overlooked this when I was reading the documentation.